src/test/java/com/datastax/oss/protocol/internal/request/AuthResponseTest.java

```diff
@@ -70,4 +70,9 @@ public void should_not_attempt_to_clear_token_if_read_only() {
     // Check that the contents are still intact
     assertThat(tokenBytes).containsExactly(0xca, 0xfe, 0xba, 0xbe);
   }
+
+  @Test
+  public void should_allow_null_token() {
+    encode(new AuthResponse(null), ProtocolConstants.Version.V4);
+  }
```

Comment:
```suggestion

  @Test
  @UseDataProvider(location = TestDataProviders.class, value = "protocolV3OrAbove")
  public void should_encode_and_decode_null_token(int protocolVersion) {
    AuthResponse initial = new AuthResponse(null);
    int encodedSize = encodedSize(initial, protocolVersion);
    MockBinaryString encoded = encode(initial, protocolVersion);
    assertThat(encoded).isEqualTo(new MockBinaryString().bytes("0x"));
    assertThat(encodedSize).isEqualTo(PrimitiveSizes.INT);
    AuthResponse decoded = decode(encoded, protocolVersion);
    assertThat(Bytes.toHexString(decoded.token)).isEqualTo("0x");
  }
```

Thank you! I think you could improve your test, see my suggestion. But you will need to change `MockPrimitiveCodec` as well since this component cannot currently process null buffers:

```
  @Override
  public void writeBytes(ByteBuffer bytes, MockBinaryString dest) {
    if (bytes == null) {
      dest.bytes("0x");
    } else {
      dest.bytes(Bytes.toHexString(bytes));
    }
  }
```

Reply:
Oh interesting, thanks will do.